File: Algorithms/Warmup/DiagonalDifference.java
package Warmup;

import java.util.Scanner;
/**
 *	Algorithms -> Warmup -> Diagonal Difference 
 *	Easy (Super Easy)
 */
public class DiagonalDifference {

	public static void main(String[] args) {
		 Scanner in = new Scanner(System.in);
	     int n = in.nextInt();
	     int a[][] = new int[n][n];
	     for(int a_i=0; a_i < n; a_i++){
	    	 for(int a_j=0; a_j < n; a_j++){
	            a[a_i][a_j] = in.nextInt();
	         }
	     }
	     in.close();
	     int diagnolDifference = computeDiagnolDifference(a);
	     System.out.println(diagnolDifference);
	}
	
	public static int computeDiagnolDifference(int a[][])
	{
		int primaryDiagnolSum = 0;
        int secondaryDiagnolSum = 0;
        int n = a.length;
		for(int a_i=0; a_i < n; a_i++){
            for(int a_j=0; a_j < n; a_j++){
                if(a_i == a_j) primaryDiagnolSum += a[a_i][a_j];
                if( (a_i + a_j) == ( n - 1) ) secondaryDiagnolSum += a[a_i][a_j];
            }
        }
        int diagnolDifference = Math.abs(primaryDiagnolSum - secondaryDiagnolSum);
		return diagnolDifference;
	}
}

File: Algorithms/Warmup/DiagonalDifference.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

/**
 *	Algorithms -> Warmup -> Diagonal Difference
 *	Easy (Super Easy)
 */
int main(){
    int n;
    scanf("%d",&n);
    int a[n][n];
    for(int a_i = 0; a_i < n; a_i++){
       for(int a_j = 0; a_j < n; a_j++){
          scanf("%d",&a[a_i][a_j]);
       }
    }

    int diagnol1 = 0;
    int diagnol2 = 0;

    for(int i = 0; i < n; i++) {
    	diagnol1 += a[i][i];
    	//printf("%d \n", a[i][i]);
    	diagnol2 += a[n - i - 1][i];
    	//printf("%d \n", a[n - i - 1][i]);
    }

    //printf("diag1 %d diag2 %d\n", diagnol1, diagnol2);
    int diagnol = abs(diagnol1 - diagnol2);

    printf("%d", diagnol);

    return 0;
}
